lib/clients/graphql/graphql_client.ts

```diff
@@ -78,13 +78,19 @@ export class GraphqlClient {
       ...params,
     });
 
-    if ((result.body as unknown as {[key: string]: unknown}).errors) {
+    // Get errors array
+    const errors = (result.body as unknown as {[key: string]: unknown})
+      .errors as any[];
+
+    // Throw error if response contains errors
+    if (errors?.length > 0) {
       throw new ShopifyErrors.GraphqlQueryError({
-        message: 'GraphQL query returned errors',
+        message: errors[0]?.message ?? 'GraphQL query returned errors',
```

Comment:
Adding a suffix like that won't be a good DX.

In my case, I'm passing down the error message to client app as graphql error and displays an error toast with this original message.

The client doesn't need to know or have access to response object. So "Refer response.errors ..." suffix will be bit misleading.

What's more important is seeing the very first original error message from that array. If there're are multiple errors, current PR will show the next error after first one is fixed.

One problem at a time 😃

Reply:
@paulomarg Fixed all tests

Reply:
Thanks! I agree that the suffix isn't really necessary, but I'd also suggest not using the exception message as an error message if the person viewing it is non-technical (e.g. a merchant), as the message might not make sense to them.

It still makes sense to return the message to developers because, as you mentioned, it makes it a lot easier to debug. I'll merge this now, but just wanted to raise the suggestion!

Reply:
I'm passing down the error to client app is to target the `preview` and `stage` environment testers. They're bit technical and reports these errors promptly.

By the time the app goes into `production`, these API errors are usually resolved, hence the merchants are less likely to see these errors.